app/models/project.rb

```diff
@@ -1,4 +1,19 @@
 class Project < ApplicationRecord
   validates_presence_of :name, :material
   belongs_to :challenge
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
+
+  def count_contestants
+    Contestant.joins(:projects).where(projects: {id: self.id}).count
```

Comment:
Nice job with this joins! I think you can also do it like this: 

```
contestants.count
```

Since the `joins` you're writing is taken care of by the association `has_many :contestants, through: :contestant_projects`. If you want to see for yourself, enter a `pry` here and type `contestants.to_sql`; you should see a similar SQL statement to what is generated from your AR query here. 

To be clear, this isn't wrong, it's just more work than is needed, thanks to the associations we have already set up.